packages/window/src/theme.rs

```diff
@@ -0,0 +1,277 @@
+//! Theme utilities.
+//!
+//! Access the system's theme to use for common tasks such as automatically setting your app styling.
+//!
+//! Most apps will need to choose a default theme in the event of an error.
+//! We recommend using either [`Result::unwrap_or`] or  [`Result::unwrap_or_default`] to do this.
+//!
+//! #### Platform Support
+//! Theme is available for Web, Windows, & Mac. Linux is unsupported and Android/iOS has not been tested.
+//!
+//! # Examples
+//! An example of using the theme to determine which class to use.
+//! ```rust
+//! use dioxus::prelude::*;
+//! use dioxus_window::theme::{use_theme, Theme};
+//!
+//! #[component]
+//! fn App() -> Element {
+//!     let theme = use_theme();
+//!     
+//!     // Default to a light theme in the event of an error.
+//!     let class = match theme().unwrap_or(Theme::Light) {
+//!         Theme::Light => "bg-light",
+//!         Theme::Dark => "bg-dark",
+//!     };
+//!
+//!     rsx! {
+//!         div {
+//!             class: "{class}",
+//!             "the current theme is: {theme().unwrap_or(Theme::Light)}"
+//!         }
+//!     }
+//! }
+//! ```
+use dioxus::prelude::*;
+use std::{error::Error, fmt::Display};
+
+/// A color theme.
+///
+/// For any themes other than `light` and `dark`, a [`ThemeError::UnknownTheme`] will be returned.
+/// We may be able to support custom themes in the future.
+#[derive(Debug, Clone, Copy, PartialEq, Default)]
+pub enum Theme {
+    /// A light theme, better in direct sunlight.
+    #[default]
+    Light,
+    /// A dark theme, better for the night owls.
+    Dark,
+}
+
+impl Display for Theme {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            Self::Light => write!(f, "light"),
+            Self::Dark => write!(f, "dark"),
+        }
+    }
+}
+
+/// Possible theme errors.
+#[derive(Debug, Clone, PartialEq)]
+pub enum ThemeError {
+    /// Theme is not supported on this platform.
+    Unsupported,
+    /// Failed to get the system theme.
+    CheckFailed,
+    /// System returned an unknown theme.
+    UnknownTheme,
+}
+
+impl Error for ThemeError {}
+impl Display for ThemeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            Self::Unsupported => write!(f, "the current platform is not supported"),
+            Self::CheckFailed => write!(
+                f,
+                "the system returned an error while checking the color theme"
+            ),
+            Self::UnknownTheme => write!(
+                f,
+                "the system provided a theme other than `light` or `dark`"
+            ),
+        }
+    }
+}
+
+type ThemeResult = Result<Theme, ThemeError>;
```

Comment:
I am surprised clippy doesn't say anything about this considering its returned by the hook
```suggestion
pub type ThemeResult = Result<Theme, ThemeError>;
```

Reply:
It's an internal helper type - the public API takes the type that it references as it's actual type while we use it wherever internally. So publicly it turns into `Result<Theme, ThemeError>` instead of `ThemeResult`. 

We can make it public if you'd want, I'm thinking that it wouldn't be used much by the consumer and figured we could keep the exports cleaner by hiding it.